process/process.go

```diff
@@ -84,6 +84,10 @@ func (p *Process) Start() error {
 		p.Pid = p.command.Process.Pid
 		p.setRunning(true)
 
+		p.mu.Lock()
+		p.done = make(chan struct{})
```

Comment:
I was just wondering why `p.Done()` would ever have to construct a channel before one was available (the if == nil bit), and the only case I could think of was to subscribe to a process finish before starting it, but initializing the channel again here would stomp on that usage...?

Reply:
Yup, agreed. I juggled the code around in the latest commit so that the channel is created earlier, but I left the check in `Done()` to support calls to `Done()` before `Start()`.